src/native/s3select/s3select_napi.cpp

```diff
@@ -0,0 +1,251 @@
+        result.Set("select", select_buf);
+
+        u_int32_t prelude[3];
+        int32_t prelude_crc, message_crc, prelude_crc_BE;
+        prelude[0] = htonl(select.length() + headers_len + 16);
+        prelude[1] = htonl(headers_len);
+        prelude_crc = crc32_gzip_refl_base(0, (u_int8_t*)prelude, 8);
+        result.Set("prelude_crc", prelude_crc);
+        prelude_crc_BE = htonl(prelude_crc);
+        /*std::cout << "sz = " << sizeof(headers_bytes) << ", native prelude_crc = " << prelude_crc << ", prelude = " << std::hex << std::setfill('0') << std::setw(2) << prelude[0] << prelude[1];
+        std::cout << std::endl;
+        std::cout.flush();*/
+        message_crc = crc32_gzip_refl_base(prelude_crc, (u_int8_t*)&prelude_crc_BE, 4);
+        message_crc = crc32_gzip_refl_base(message_crc, const_cast<u_int8_t*>(headers_bytes), headers_len);
+        message_crc = crc32_gzip_refl_base(message_crc, (u_int8_t*)select.data(), select.length());
+        result.Set("message_crc", message_crc);
+
+        deferred.Resolve(result);
+    }
+
+    void OnError(Napi::Error const& error) override
+    {
+        Napi::Env env = Env();
+        auto obj = error.Value();
+        deferred.Reject(obj);
+    }
+};
+
+class S3SelectNapi : public Napi::ObjectWrap<S3SelectNapi>
+{
+
+public:
+    static Napi::Object Init(Napi::Env env, Napi::Object exports);
+    S3SelectNapi(const Napi::CallbackInfo& info);
+    Napi::Value Write(const Napi::CallbackInfo& info);
+    Napi::Value Flush(const Napi::CallbackInfo& info);
+    ~S3SelectNapi();
+
+private:
+    static Napi::FunctionReference constructor;
+    InputFormat input_format;
+    s3selectEngine::s3select s3select;
+    s3selectEngine::csv_object* csv_object = nullptr;
+    s3selectEngine::json_object* json_object = nullptr;
+    const u_int8_t* headers_buf;
+    u_int32_t headers_len;
+};
+
+Napi::Value
+S3SelectNapi::Write(const Napi::CallbackInfo& info)
+{
+    Napi::Buffer<char> buffer = info[0].As<Napi::Buffer<char>>();
+
+    /*std::cout << "write len = " << buffer.Length() <<" data = " << buffer.Data() << std::endl;
+    std::cout.flush();*/
+
+    SelectWorker* worker = new SelectWorker(info.Env(), *csv_object, *json_object, buffer.Data(), buffer.Length(), false, input_format, headers_buf, headers_len);
+    worker->Queue();
+    return worker->deferred.Promise();
+}
+
+Napi::Value
+S3SelectNapi::Flush(const Napi::CallbackInfo& info)
+{
+    SelectWorker* worker = new SelectWorker(info.Env(), *csv_object, *json_object, nullptr, 0, true, input_format, headers_buf, headers_len);
+    worker->Queue();
+    return worker->deferred.Promise();
+}
+
+Napi::FunctionReference S3SelectNapi::constructor;
+
+Napi::Object
+S3SelectNapi::Init(Napi::Env env, Napi::Object exports)
+{
+    Napi::HandleScope scope(env);
+
+    Napi::Function func = DefineClass(env, "S3SelectNapi", { InstanceMethod("write", &S3SelectNapi::Write), InstanceMethod("flush", &S3SelectNapi::Flush) });
```

Comment:
add line breaks for readability -
```suggestion
    Napi::Function func = DefineClass(env, "S3SelectNapi", {
        InstanceMethod("write", &S3SelectNapi::Write),
        InstanceMethod("flush", &S3SelectNapi::Flush),
    });
```